### PR TITLE
DEVOPS-816_kubescape_on_push_resolution

### PR DESCRIPTION
**Pull request by APB-s**
## What 🕵️ 🔍

- The scan will on a specific path only when a file has been changed in it.

## References 📝 🔗

- [Jira task](https://xainag.atlassian.net/browse/)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Update(s) to the configuration of existing service(s)
- [ ] This change requires a documentation update
- [ ] Removed some deprecated application or functionality

## Reviewer checklist

*Reviewer agreement:*

* Reviewers shouldn't review a "Draft Pull Request", but rather wait for it to be marked as "Ready for Review".
* Reviewers do **not** commit or merge the pull request.
* If/when there are changes that need to be applied (applying Terraform or deploying Kubernetes manifests), the **assignee** is the one responsible for doing it.
* Use the labels `not-applied`, `partially-applied` and `fully-applied` so others know the current progress.
  * Remember to change labels accordingly once you start and finish to apply the changes.

## Checklist:

- [x] I have made corresponding changes to the documentation.
- [x] I have performed a self-review of my own code.
- [x] I did not commit anything sensitive like keys or secrets.
- [x] I added the corresponding labels.
- [x] I set the reviewers and the assignees.
- [ ] All open comments and discussions have been resolved.
- [x] My branch is following the name pattern: `JIRA-ID_Short_descripton`, example: `DEVOPS-000_some_meaningful_description`

### CODE REVIEW

**Comment by APB-s** [comment]
I need to merge to check about the scan problem, and see if it is because of merging that the scan doesn't work

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -40,55 +36,79 @@ jobs:
...
 
-  ci-ok:
-    name: ci-ok
-    needs:
-      # - kubeval-check
-      - nsa-security-check
-    runs-on: ubuntu-20.04
-    steps:
-      - run: exit 0
+  # ci-ok:
```
Comment (abargiela) [inline .github/workflows/ci.yaml:107]: Do we need these commented lines here? Can we erase them?

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -1,37 +1,33 @@
...
-      - 'azure/kubernetes/**'
-
-concurrency:
-  group: ${{ github.ref }}
-  cancel-in-progress: true
-   
+        - 'ionos/kubernetes/**'
+        - 'azure/kubernetes/**'
 jobs:
-  # kubeval-check:
```
Comment (jorge-cr-13) [inline .github/workflows/ci.yaml:14]: Why are you getting rid of this?

File: .github/workflows/ci.yaml [hunk 3]
```diff
@@ -40,55 +36,47 @@ jobs:
...
         run: curl -s https://raw.githubusercontent.com/armosec/kubescape/master/install.sh | /bin/bash
 
-      # Scanning development
-      - name: Scan DEV YAML files
-        run: kubescape scan framework nsa -t 30 ionos/kubernetes/dev/*.yaml 
-
       # Scanning DE Cluster
       - name: Scan DE YAML files
-        run: kubescape scan framework nsa -t 30 ionos/kubernetes/prod/de/*.yaml 
+        run: kubescape scan framework nsa -t 30 --enable-host-scan ionos/kubernetes/prod/*/*.yaml
```
Comment (jorge-cr-13) [inline .github/workflows/ci.yaml:41]: Why is the `--enable-host-scan` parameter needed?

File: .github/workflows/ci.yaml [hunk 4]
```diff
@@ -40,55 +36,47 @@ jobs:
...
-        run: kubescape scan framework nsa -t 30 azure/kubernetes/prod/se/as/*.yaml 
+      - name: Install Kubescape
+        run: curl -s https://raw.githubusercontent.com/armosec/kubescape/master/install.sh | /bin/bash
       
-      # Scanning UAE
-      - name: Scan UAE YAML files
-        run: kubescape scan framework nsa -t 30 azure/kubernetes/prod/uae/n/*.yaml 
+      # Scanning AU
+      - name: Scan YAML files
+        run: kubescape scan framework nsa -t 70 --enable-host-scan azure/kubernetes/prod/*/*/*.yaml
```
Comment (jorge-cr-13) [inline .github/workflows/ci.yaml:73]: Why this is 70 and the others 30?

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (APB-s) [reply .github/workflows/ci.yaml:14]: I delete it because we don't use it.

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (APB-s) [reply .github/workflows/ci.yaml]: No we don't need it, it was for testing as I didn't finish yet.

File: azure/kubernetes/prod/au/e/ns_production/haproxy/base/kustomization.yaml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
-
+#test3
```
[the remark this answers is not in the document]
Comment (APB-s) [reply azure/kubernetes/prod/au/e/ns_production/haproxy/base/kustomization.yaml:3]: No we don't need it, it was for testing as I didn't finish yet. I'll delete it

File: ionos/kubernetes/dev/de/txl/ns_development/haproxy/base/kustomization.yaml [hunk 6]
```diff
@@ -5,7 +5,7 @@ commonLabels:
   app.kubernetes.io/component: haproxy
   app.kubernetes.io/name: api-gw
   app.kubernetes.io/part-of: xayn
-
+  #test3
```
[the remark this answers is not in the document]
Comment (APB-s) [reply ionos/kubernetes/dev/de/txl/ns_development/haproxy/base/kustomization.yaml]: Not necessary, I'll make the change too

**Pull request merged**